crates/cext/build_header.rs

```diff
@@ -0,0 +1,27 @@
+// This code is part of Qiskit.
+//
+// (C) Copyright IBM 2025
+//
+// This code is licensed under the Apache License, Version 2.0. You may
+// obtain a copy of this license in the LICENSE.txt file in the root directory
+// of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+//
+// Any modifications or derivative works of this code must retain this
+// copyright notice, and modified files need to carry a notice indicating
+// that they have been altered from the originals.
+
+extern crate cbindgen;
+
+/// This function generates the C header for Qiskit from the qiskit-cext crate.
+fn main() {
+    // Pull the config from the cbindgen.toml file.
+    let config = cbindgen::Config::from_file("cbindgen.toml").unwrap();
+
+    // Build the header.
+    cbindgen::Builder::new()
+        .with_crate(".")
+        .with_config(config)
+        .generate()
+        .expect("Unable to generate C bindings.")
+        .write_to_file("qiskit.h");
+}
```

Comment:
Personally I would say that it's not good behaviour to have a build script write a dirty built file into the source repo as a matter of habit.  Artifacts built by `cargo` should probably go in Cargo's output directory `/target/.../`, and the `Makefile` version can "install" that into the distribution location.

Imo, only a manual call to the command-line tool should place dirty files in the CWD (if we even keep the command-line tool as a stand-alone thing).

Reply:
Updated to write `qiskit.h` into `target/qiskit.h` as per out offline discussion